pinot-core/src/main/java/org/apache/pinot/core/transport/ServerResponse.java

```diff
@@ -50,6 +51,14 @@ public int getSubmitDelayMs() {
     }
   }
 
+  public int getRequestSentDelayMs() {
+    if (_sendRequestTimeMs != 0) {
+      return (int) (_sendRequestTimeMs - _startTimeMs);
+    } else {
+      return -1;
+    }
```

Comment:
@jackjlli , this seems incorrect. this is for time it took to send request on transport so although you are emitting in the metrics correctly , here since the delta is computed from start time, it gets recorded incorrectly

Reply:
The following is correct for existing `getSubmitDelayMs()` API

```
public int getSubmitDelayMs() {                          
  if (_submitRequestTimeMs != 0) {                       
    return (int) (_submitRequestTimeMs - _startTimeMs);  
  } else {                                               
    return -1;                                           
  }                                                      
}  
```

But what you are measuring is delay it took on the wire `getRequestSentDelayMs()`, so you should no rely on _startTime. Simply change the API to record the time


```
void markRequestSent(ServerRoutingInstance serverRoutingInstance, long timeMs) {
    _responseMap.get(serverRoutingInstance).markRequestSent(timeMs);
  }
```